app/services/folio_client.rb

```diff
@@ -224,6 +245,30 @@ def locations
 
   private
 
+  # Find a user by barcode in FOLIO; raise an error if not found
+  def find_user_by_barcode(barcode)
+    user = get_json('/users', params: { query: CqlQuery.new(barcode:).to_query }).dig('users', 0)
+    raise ActiveRecord::RecordNotFound, "User with barcode '#{barcode}' not found" unless user
+
+    user
+  end
+
+  # Find a user by university ID (externalSystemId in FOLIO); raise an error if not found
+  def find_user_by_university_id(university_id)
+    user = get_json('/users', params: { query: CqlQuery.new(externalSystemId: university_id).to_query }).dig('users', 0)
+    raise ActiveRecord::RecordNotFound, "User with externalSystemId '#{university_id}' not found" unless user
+
+    user
+  end
+
+  # Find a user by sunetid (username in FOLIO); raise an error if not found
+  def find_user_by_sunetid(sunetid)
+    user = get_json('/users', params: { query: CqlQuery.new(username: sunetid).to_query }).dig('users', 0)
+    raise ActiveRecord::RecordNotFound, "User with username '#{sunetid}' not found" unless user
+
+    user
+  end
+
```

Comment:
these were repeated across the `login_by_*` and `find_patron_by_*` methods, so I extracted them.